kernel/src/hil/sensors.rs

```diff
@@ -31,6 +31,18 @@ pub trait HumidityClient {
     fn callback(&self, value: usize);
 }
 
+/// A basic interface for a proximity sensor
+pub trait ProximityDriver<'a> {
+    fn set_client(&self, client: &'a dyn ProximityClient);
+    fn read_proximity(&self) -> ReturnCode; // Instantaneous proximity reading
```

Comment:
It is a little misleading to call this instantaneous, as it is still asynchronous (values are returned in a callback), and the time for this to return probably depends on the underlying driver.

Reply:
Hi @hudson-ayers, no problem and sorry for the delay!  I have addressed all of your comments but feel free to check if I have accidentally missed one.  I have provided better comments to detail the interface to the user.

Reply:
"comments" as in comments in the code